Add dateOfBirth property to user and its requested pipes

mgandore opened the pull request.

mgandore committed:
Add dateOfBirth prop to users

Modify mothods of user-list.component and user-details template
in consequence of the new added prop.
*TODO: Implement pipe to display age.

mgandore committed:
Implement userAge pipe

Filters the date of birth of each user and displays his age.

mgandore committed:
Another pipe to filer adult users

Backgrond color of each adult user sHoUlD be green.

danielursu commented on component-interaction/src/app/isAdultUser.pipe.ts:
```diff
@@ -0,0 +1,11 @@
+	name: "isAdult"
+})
+
+export class isAdultUserPipe implements PipeTransform {
```
```suggestion
export class IsAdultUserPipe implements PipeTransform {

```

danielursu commented on component-interaction/src/app/isAdultUser.pipe.ts:
```diff
@@ -0,0 +1,11 @@
+@Pipe({
+	name: "isAdult"
+})
+
```
```suggestion
```

danielursu commented on component-interaction/src/app/user-list/user-list.component.ts:
```diff
@@ -63,12 +67,24 @@ export class UserListComponent implements OnInit {
+			return Math.floor(Math.random() * 12) + 1;
+		}
+		return Math.floor(Math.random() * 31) + 1;
+
+
```
```suggestion
```

danielursu commented on component-interaction/src/app/user-list/user-list.component.ts:
```diff
@@ -63,12 +67,24 @@ export class UserListComponent implements OnInit {
 	}
 
+	public getRandomNumForDate(input?: string): number {
+		if (input === "year") {
+			return Math.floor(Math.random() * 100) + 1900;
+		} else if (input === "month") {
+			return Math.floor(Math.random() * 12) + 1;
+		}
+		return Math.floor(Math.random() * 31) + 1;
```
use switch + InputDateEnum

danielursu commented on component-interaction/src/app/userAge.pipe.ts:
```diff
@@ -0,0 +1,10 @@
+})
+export class UserAgePipe implements PipeTransform {
+	public transform(dateOfBirth: any, ...args: any[]): number {
+		return 2022 - dateOfBirth?.getFullYear();
```
```suggestion
		return 2022 - dateOfBirth?.getFullYear();

```
will this snippet work next year? :))

mgandore replied:
😭😭😭🤦‍♂️

mgandore committed:
Make requested changes change approach on styling.

-Typos, empty lines;
-Refactor getRandomNumForDate() function;
-Make userAge.pipe work correctly even next year;

danielursu commented on component-interaction/src/app/user-list/user-list.component.ts:
```diff
@@ -63,12 +69,32 @@ export class UserListComponent implements OnInit {
 
+	public getRandomNumForDate(input: InputDateEnum): number {
+		let output: number = 0;
+		switch (input) {
+			case "year":
+				output = Math.floor(Math.random() * 100) + 1900;
+				break;
```
```suggestion
		switch (input) {
			case InputDateEnum.YEAR:
				return Math.floor(Math.random() * 100) + 1900;

```

danielursu commented on component-interaction/src/app/user-list/user-list.component.ts:
```diff
@@ -63,12 +69,32 @@ export class UserListComponent implements OnInit {
+				output = Math.floor(Math.random() * 31) + 1;
+				break;
+		}
+		return output;
```
same here

danielursu replied:
also you can get rid of output variable and return 0 after switch

mgandore committed:
Another empty line

danielursu commented on component-interaction/src/app/userAge.pipe.ts:
```diff
@@ -0,0 +1,11 @@
+})
+export class UserAgePipe implements PipeTransform {
+	public transform(dateOfBirth: any, ...args: any[]): number {
+		let currentDate = new Date();
+		return currentDate.getFullYear() - dateOfBirth.getFullYear();
```
```suggestion
		return new Date().getFullYear() - dateOfBirth.getFullYear();

```

danielursu commented on component-interaction/src/app/app.module.ts:
```diff
@@ -2,16 +2,20 @@ import { NgModule } from '@angular/core';
 import { SortPipe } from './sort.pipe';
 import { UserDetailsComponent } from './user-details/user-details.component';
 import { UserListComponent } from './user-list/user-list.component';
+import { UserAgePipe } from './userAge.pipe';
```
double qoutes

danielursu commented on component-interaction/src/app/inputDate.enum.ts:
```diff
@@ -0,0 +1,5 @@
+export enum InputDateEnum {
+	year = "year",
+	month = "month",
+	day = "day"
```
```suggestion
	YEAR = "year",
	MONTH = "month",
	DAY = "day"

```

danielursu commented on component-interaction/src/app/isAdultUser.pipe.ts:
```diff
@@ -0,0 +1,10 @@
+import { Pipe, PipeTransform } from "@angular/core";
```
is-adult-user.pipe.ts

danielursu commented on component-interaction/src/app/userAge.pipe.ts:
```diff
@@ -0,0 +1,11 @@
+import { Pipe, PipeTransform } from "@angular/core";
```
user-age.pipe.ts

mgandore committed:
Make requested changes #2

-Renamed some files in kebab-case-style;
-More refactoring;
-Replace single quotes do double ones;
